KEYCLOAK-12904 Add Moment.js info, run download script

ASzc opened the pull request:
Added and downloaded for both rh-sso and keycloak directories.

ASzc committed:
KEYCLOAK-12904 Add Moment.js info, run download script

Also fix misc. html license files

vmuzikar commented on distribution/feature-packs/server-feature-pack/src/main/resources/licenses/rh-sso/licenses.xml:
```diff
@@ -903,6 +903,18 @@
+          <url>https://raw.githubusercontent.com/moment/moment/develop/LICENSE</url>
+        </license>
+      </licenses>
+    </other>
```
I think we need to do the same for Keycloak as well (the other licenses.xml).

ASzc replied:
It should be there:

```
aszczucz@tyrfing ~/s/k/keycloak3 (KEYCLOAK-12904)> git grep Moment.js
distribution/feature-packs/server-feature-pack/src/main/resources/licenses/keycloak/licenses.xml:      <description>Moment.js</description>
distribution/feature-packs/server-feature-pack/src/main/resources/licenses/rh-sso/licenses.xml:      <description>Moment.js</description>
```

vmuzikar reviewed (changes requested):
@ASzc Thanks for the quick update!

vmuzikar reviewed (dismissed):
@ASzc Ah, right, sorry – I somehow missed that. LGTM, thanks!

martin-kanis commented on distribution/feature-packs/server-feature-pack/src/main/resources/licenses/rh-sso/Emotion,MIT License.txt:
```diff
@@ -0,0 +1,41 @@
+{
```
JSON in txt file

ASzc replied:
This was the result of no license file existing in the repo. The others are not intentional, and I'll have a look at them.

martin-kanis replied:
You can probably use https://raw.githubusercontent.com/emotion-js/emotion/master/LICENSE

martin-kanis commented on distribution/feature-packs/server-feature-pack/src/main/resources/licenses/rh-sso/ChangeCase,MIT License.txt:
```diff
@@ -0,0 +1,1054 @@
+
```
HTML in txt file

martin-kanis commented on distribution/feature-packs/server-feature-pack/src/main/resources/licenses/keycloak/Patternfly 4 React,MIT License.txt:
```diff
@@ -0,0 +1,1065 @@
+
```
HTML in txt file.

martin-kanis commented on distribution/feature-packs/server-feature-pack/src/main/resources/licenses/keycloak/Emotion,MIT License.txt:
```diff
@@ -0,0 +1,41 @@
+{
```
JSON in txt file

martin-kanis replied:
You can probably use https://raw.githubusercontent.com/emotion-js/emotion/master/LICENSE

martin-kanis commented on distribution/feature-packs/server-feature-pack/src/main/resources/licenses/keycloak/ChangeCase,MIT License.txt:
```diff
@@ -0,0 +1,1054 @@
+
```
HTML in txt file

martin-kanis commented on distribution/feature-packs/server-feature-pack/src/main/resources/licenses/rh-sso/Patternfly 4 React,MIT License.txt:
```diff
@@ -0,0 +1,1082 @@
+
```
HTML again

martin-kanis reviewed (changes requested):
Some txt files contain HTML or JSON, see inline. Also there are no versions in links to the licenses. Do we want to include always latest versions?

ASzc commented:
Sorry for the sloppiness here, guys. Thanks for reviewing.

I've corrected the problems @martin-kanis identified, and also added some checks into the download script to prevent some of these from happening again.

I also had to fix the URL for rfc4648.js to pass these new checks.

ASzc commented:
Found a couple more HTML ones to fix, sisu.inject and sisu.plexus. Didn't add a rule to the script, since it's not as clear a pattern as it is with the GitHub ones.

vmuzikar reviewed (changes requested):
@martin-kanis That's a very good catch, thank you!

@ASzc I think there are more HTML files:
* aopalliance,aopalliance,1.0,Public Domain.txt
* org.eclipse.aether,aether-api,1.1.0,Eclipse Public License 1.0.txt
* org.eclipse.aether,aether-connector-basic,1.1.0,Eclipse Public License 1.0.txt
* org.eclipse.aether,aether-impl,1.1.0,Eclipse Public License 1.0.txt
* org.eclipse.aether,aether-spi,1.1.0,Eclipse Public License 1.0.txt
* org.eclipse.aether,aether-transport-file,1.1.0,Eclipse Public License 1.0.txt
* org.eclipse.aether,aether-transport-http,1.1.0,Eclipse Public License 1.0.txt
* org.eclipse.aether,aether-transport-wagon,1.1.0,Eclipse Public License 1.0.txt
* org.eclipse.aether,aether-util,1.1.0,Eclipse Public License 1.0.txt

Some of them might be intentionally HTMLs but in that case their file extensions shouldn't be .txt.

I wonder if this is getting out of scope of this PR.

ASzc commented:
I was able to trivially make the aether ones plain text.

I'd consider fixing aopalliance out of scope for this PR, since the license truely is only declared on the project's HTML splash page. I've left it as is.

vmuzikar reviewed (approved):
Thanks @ASzc!
Created https://issues.redhat.com/browse/KEYCLOAK-12927 for aopalliance

martin-kanis reviewed (changes requested):
@ASzc Thank you for the fixes. Except rfc4648.js there's two more licenses with json inside. Maybe we can use https://raw.githubusercontent.com/emotion-js/emotion/master/LICENSE for those.

vmuzikar commented:
> @ASzc Thank you for the fixes. Except rfc4648.js there's two more licenses with json inside.

@martin-kanis I thought the JSON was intentional(?)

martin-kanis commented:
@vmuzikar Json in rfc4648.js is intentional but for emotion-js/emotion there's a license file in the repository which can be used instead.

ASzc commented:
There isn't a license file in the emotion-js repo at `v9.2.12`, the tag we're using.

martin-kanis commented:
OK approving
